R/package.R

```diff
@@ -50,6 +50,10 @@
 argufy <- function(fun, ...) {
   if (!is.function(fun)) stop("'fun' must be a function")
 
+  # these statements are needed to get S4 functions to work properly
+  was_s4 <- isS4(fun)
+  old_attributes <- attributes(fun)
+
```

Comment:
If the code can be printed easily then it is fine as it is now. I'll just add it to the docs. This confused me until I debugged `argufy()`, because it did not seem to do anything. :)